addon/components/date.js

```diff
@@ -0,0 +1,62 @@
+import Component from '@ember/component';
+import layout from '../templates/components/date';
+import moment from 'moment';
+
+const MONTHS = [
+  'January',
+  'February',
+  'March',
+  'April',
+  'May',
+  'June',
+  'July',
+  'August',
+  'September',
+  'October',
+  'November',
+  'December',
+];
+
+// TO DO: make this more flexible
+const YEARS = Array(...Array(100)).map((_, i) => `${i + 1940}`);
+
+export default Component.extend({
+  layout,
+  classNames: ['cs-component-date'],
+  attributeBindings: ['dataTestName:data-test-cs-component-date'],
+  dataTestName: true,
+  months: MONTHS,
+  years: YEARS,
+
```

Comment:
You should wrap these actions in the `actions` hook. For example:

```js
actions: {
  changeSelected(ev) {
    // ...
  },

  changeCenter(unit, calendar, val) {
    // ...
  }
}
```

Then when you call the action, be sure to wrap it in quotes i.e. `{{action "changeSelected"}}`

You can read more here: https://guides.emberjs.com/release/templates/actions/

Reply:
When should we use this type of syntax to call a function as opposed to the one without the actions wrapper? I've seen both styles, but not sure what the difference is.

Reply:
I think whenever a function is triggered by any user action, we should use the `action` hook. For anything else like initial state setup, we should use regular functions. Oh and also ember-concurrency function generators don't use the action hook even though those can sometimes be triggered by user actions.